From 5fae0c2abbd30a6662a7ce9ba10fe3a9c4859969 Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Tue, 18 Dec 2018 21:25:46 +0100
Subject: [PATCH] Disable unsafe symlink check HYP-19186

If the environment variable 'EXTRACT_UNSAFE_SYMLINKS' is not set, the
unsafe symlink check is not executed.
This restores the previous behavior without the environment variable.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 archival/libarchive/unsafe_symlink_target.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/archival/libarchive/unsafe_symlink_target.c b/archival/libarchive/unsafe_symlink_target.c
index ee46e28..9f7b11d 100644
--- a/archival/libarchive/unsafe_symlink_target.c
+++ b/archival/libarchive/unsafe_symlink_target.c
@@ -18,6 +18,8 @@ unsafe:
 				return 0; /* pretend it's safe */
 			return 1; /* "UNSAFE!" */
 		}
+		return 0; /* If "EXTRACT_UNSAFE_SYMLINKS" is not set, then old behavior is used */
+#if 0
 		bb_error_msg("skipping unsafe symlink to '%s' in archive,"
 			" set %s=1 to extract",
 			target,
@@ -26,6 +28,7 @@ unsafe:
 		/* Prevent further messages */
 		setenv("EXTRACT_UNSAFE_SYMLINKS", "0", 0);
 		return 1; /* "UNSAFE!" */
+#endif
 	}
 
 	dot = target;
@@ -45,4 +48,4 @@ unsafe:
 			/* NB: it can even be trailing ".", should only add 1 */
 			dot += 1;
 	}
-}
\ No newline at end of file
+}
-- 
2.16.4

